go-controller/pkg/ovn/base_event_handler.go

```diff
@@ -55,7 +56,9 @@ func (h *baseNetworkControllerEventHandler) areResourcesEqual(objType reflect.Ty
 		if !ok {
 			return false, fmt.Errorf("could not cast obj2 of type %T to *knet.NetworkPolicy", obj2)
 		}
-		return reflect.DeepEqual(np1, np2), nil
+		areEqual := apiequality.Semantic.DeepEqual(np1.Spec, np2.Spec) &&
+			np1.Annotations[ovnStatelessNetPolAnnotationName] == np2.Annotations[ovnStatelessNetPolAnnotationName]
+		return areEqual, nil
```

Comment:
Have you considered MultiNetworkPolicies as well? Line 103 below?
I know it's not directly what you are addressing but we basically do equivalent processing for them and if you don't address it here we would have to unavoidably follow up. 
In that case, you would additionaly watch out for the `const PolicyForAnnotation = "k8s.v1.cni.cncf.io/policy-for"` annotation.